lib/galaxy/workflow/modules.py

```diff
@@ -697,6 +700,63 @@ def get_runtime_inputs(self, **kwds):
+                if connection.input_name not in tool_inputs and "|" in connection.input_name:
+                    param_name, case_name = connection.input_name.split("|", 1)
+                    if param_name in tool_inputs.data:
+                        param = tool_inputs.data[param_name]
+                    elif re.match('.+_\\d+', param_name):  # check for repeat label
+                        param_name, param_idx = param_name.rsplit('_', 1)
+                        param = tool_inputs.data[param_name]
+                    while param.type in ("conditional", "repeat"):
+                        if "|" in case_name:
+                            param_name, case_name = case_name.split("|", 1)
+                            if param.type == "conditional":
+                                sel_cases = [case for case in param.cases if param_name in case.inputs.data]
+                                if sel_cases:
+                                    sel_case = sel_cases[0]
+                                elif re.match('.+_\\d+', param_name):  # check for repeat label
+                                    param_name, param_idx = param_name.rsplit('_', 1)
+                                    sel_case = [case for case in param.cases if param_name in case.inputs.data][0]
+                                param = sel_case.inputs.data[param_name]
+                            elif param.type == "repeat":
+                                if param_name in param.inputs:
+                                    param = param.inputs[param_name]
+                                elif re.match('.+_\\d+', param_name):  # check for repeat label
+                                    param_name, param_idx = param_name.rsplit('_', 1)
+                                    param = param.inputs[param_name]
+                        else:
+                            if param.type == "conditional":
+                                sel_case = [case for case in param.cases if case_name in case.inputs.data][0]
+                                input_data = sel_case.inputs.data[case_name]
+                            elif param.type == "repeat":
+                                input_data = param.inputs[case_name]
+                            break
+                    param = tool_inputs.data[param_name]
+                    if param.type == "conditional":
+                        sel_case = [case for case in param.cases if case_name in case.inputs.data][0]
+                        input_data = sel_case.inputs.data[case_name]
+                else:
+                    input_data = tool_inputs.data[connection.input_name]
+                static_options.append(input_data.get_options(self.trans, {}))  # Aggregation input select options from several connections
+            if static_options:
+                # Intersection based on values.
+                intxn_vals = set.intersection(*({option[1] for option in options}) for options in static_options))
```

Comment:
You've got a syntax error here, but it's hard to spot.
```suggestion
                intxn_vals = set.intersection(*({option[1] for option in options} for options in static_options))
```